### PR TITLE
Natalie's code review suggestions.

### PR DESCRIPTION
**Pull request by forbesne**
Analysis of the program:
* This project searches for the best places to fly a drone and will alert the user of bad weather conditions based off of the user's current location.

Was the program available in UC Github on time?
* Yes

Is the program documented/commented well enough for you to understand?
* Yes

Does the program compile?
* Yes

What I learnt from this code review:
* I learned how to convert java code to kotlin while looking at the MainActivity.java and MapsActivityCurrentPlace.kt
* I learned how to set and adjust opacity from TransparentTileOWM.kt
* I learned about try catch statements in kotlin from TransparentTileOWM.kt and doing extra research about it for this code review

### CODE REVIEW

File: app/src/main/java/edu/uc/zhukv/droneradarmap/Weather_Layer/TransparentTileOWM.kt [hunk 1]
```diff
@@ -14,11 +14,14 @@ import java.net.URL
...
 
+    init {
+        setOpacity(50)
+    }
     /**
      * Sets the desired opacity of map [Tile]s, as a percentage where 0% is invisible and 100% is completely opaque.
      * @param opacity The desired opacity of map [Tile]s (as a percentage between 0 and 100, inclusive)
      */
-    fun setOpacity(opacity: Int) {
+    internal fun setOpacity(opacity: Int) {
```
Comment (forbesne) [inline app/src/main/java/edu/uc/zhukv/droneradarmap/Weather_Layer/TransparentTileOWM.kt:36]: Updated visibility modifiers as recommended from: https://kotlinlang.org/docs/visibility-modifiers.html#classes-and-interfaces

File: app/src/main/java/edu/uc/zhukv/droneradarmap/Weather_Layer/TransparentTileOWM.kt [hunk 2]
```diff
@@ -14,11 +14,14 @@ import java.net.URL
 class TransparentTileOWM(private val tileType: String):TileProvider {
     private val opacityPaint = Paint()
 
+    init {
+        setOpacity(50)
+    }
```
Comment (forbesne) [inline app/src/main/java/edu/uc/zhukv/droneradarmap/Weather_Layer/TransparentTileOWM.kt:20]: Moved the init block up as recommended from: https://kotlinlang.org/docs/coding-conventions.html#class-layout

File: app/src/main/java/edu/uc/zhukv/droneradarmap/MapsActivityCurrentPlace.kt [hunk 3]
```diff
@@ -18,6 +18,7 @@ class MapsActivityCurrentPlace(var NameOfCity: TextView? = null) {
     val MIN_TIME: Long = 5000
     val MIN_DISTANCE = 1000f
     val REQUEST_CODE = 101
+    private val PERMISSIONS_REQUEST_ACCESS_FINE_LOCATION: Int = 0
```
Comment (forbesne) [inline app/src/main/java/edu/uc/zhukv/droneradarmap/MapsActivityCurrentPlace.kt]: Moved constant declaration to be accessible in multiple functions and should not be modified after initialization as recommended from: https://kotlinlang.org/docs/coding-conventions.html#idiomatic-use-of-language-features

File: app/src/main/java/edu/uc/zhukv/droneradarmap/MapsActivityCurrentPlace.kt [hunk 4]
```diff
@@ -43,17 +44,17 @@ class MapsActivityCurrentPlace(var NameOfCity: TextView? = null) {
...
             }
 
 
-    fun onRequestPermissionsResult(requestCode: Int,
+    internal fun onRequestPermissionsResult(requestCode: Int,
                                             permissions: Array<String>,
                                             grantResults: IntArray) {
         var locationPermissionGranted = false
-        val PERMISSIONS_REQUEST_ACCESS_FINE_LOCATION = null
+
```
Comment (forbesne) [inline app/src/main/java/edu/uc/zhukv/droneradarmap/MapsActivityCurrentPlace.kt:87]: Removed null assignment to variable of non-null type, (new constant declaration at top), as recommended from: https://kotlinlang.org/docs/null-safety.html

File: app/src/main/java/edu/uc/zhukv/droneradarmap/MapsActivityCurrentPlace.kt (hunk 3 above)
Comment (discospiff) [reply app/src/main/java/edu/uc/zhukv/droneradarmap/MapsActivityCurrentPlace.kt]: Good.

File: app/src/main/java/edu/uc/zhukv/droneradarmap/MapsActivityCurrentPlace.kt (hunk 4 above)
Comment (discospiff) [reply app/src/main/java/edu/uc/zhukv/droneradarmap/MapsActivityCurrentPlace.kt:87]: Very good.

File: app/src/main/java/edu/uc/zhukv/droneradarmap/MapsActivityCurrentPlace.kt [hunk 5]
```diff
@@ -43,17 +44,17 @@ class MapsActivityCurrentPlace(var NameOfCity: TextView? = null) {
...
             } else {
-                val PERMISSIONS_REQUEST_ACCESS_FINE_LOCATION = 0
+
                 //ActivityCompat.requestPermissions(this, arrayOf(Manifest.permission.ACCESS_FINE_LOCATION),
                         //PERMISSIONS_REQUEST_ACCESS_FINE_LOCATION)
             }
 
 
-    fun onRequestPermissionsResult(requestCode: Int,
+    internal fun onRequestPermissionsResult(requestCode: Int,
```
Comment (discospiff) [inline app/src/main/java/edu/uc/zhukv/droneradarmap/MapsActivityCurrentPlace.kt]: internal access is better than default.

**Comment by discospiff** [comment]
Several good suggestions here that align  with best practices and reduce technical debt.

The team should discuss and consider merging.

**Pull request merged**